src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs

```diff
@@ -128,11 +128,11 @@ private protected override void CheckScoreAndLabelTypes(RoleMappedSchema schema)
             var host = Host.SchemaSensitive();
             var t = score.Type;
             if (t != NumberType.Float)
-                throw host.ExceptSchemaMismatch(nameof(schema), "score", score.Name, "R4", t.ToString());
+                throw host.ExceptSchemaMismatch(nameof(schema), "score", score.Name, "float", t.ToString());
             host.Check(schema.Label.HasValue, "Could not find the label column");
             t = schema.Label.Value.Type;
             if (t != NumberType.R4 && t != NumberType.R8 && t != BoolType.Instance && t.GetKeyCount() != 2)
-                throw host.ExceptSchemaMismatch(nameof(schema), "label", schema.Label.Value.Name, "R4, R8, BL or a 2-value key", t.ToString());
+                throw host.ExceptSchemaMismatch(nameof(schema), "label", schema.Label.Value.Name, "float, double, bool, or a 2-value KeyType", t.ToString());
```

Comment:
>label [](start = 65, length = 5)

Instead of having the literal string "label" in each of these SchemaMismatch exceptions, what about adding some common column names somewhere as const strings? Label, Score, Probability, and Features seem like some common ones to use.

Reply:
also what is a 2-value KeyType?

---
In reply to: [251127632](https://github.com/dotnet/machinelearning/pull/2217#discussion_r251127632) [](ancestors = 251127632)

Reply:
The problem with DefaultColumnNames is twofold:

1. It contains "Column" in the name (which would lead to a repetition of column in the message)
2. It does not contain many column roles such the very common "input" case.

In general, my understanding from talking with Tom is that the intent of the `columnRole` parameter was to be a interpretable qualification of the column role. 

Quoting Tom: 

"
I think Pete's intention was that it would be a descriptive human readable name.
So I've been going with hard coded small things like "input" and "label" and stuff.
"

---
In reply to: [251130649](https://github.com/dotnet/machinelearning/pull/2217#discussion_r251130649) [](ancestors = 251130649,251127632)

Reply:
will fix the 2-value KeyType thanks! I'll check everywhere for that error.

---
In reply to: [251143821](https://github.com/dotnet/machinelearning/pull/2217#discussion_r251143821) [](ancestors = 251143821,251130649,251127632)

Reply:
Sorry - when you say it contains Column, are you referring to the variable name DefaultColumnNames? Is the concern this would look weird from the coding perspective? To the user, they willl just see the column name like "Label". And I think its good to have one place where this string is used as this makes it easier to maintain. I can talk more about it offline if you want.

---
In reply to: [251143821](https://github.com/dotnet/machinelearning/pull/2217#discussion_r251143821) [](ancestors = 251143821,251130649,251127632)

Reply:
I used DefaultColumnNames when possible (for the Features, Score, GroupID, Label).